website/utils/url_utils.ts

```diff
@@ -2,16 +2,60 @@ import FirebaseRequest, { DatasetMetaData } from "../../public/firebase";
 
 import { AppProps, GlobalViewerSettings } from "../../src/aics-image-viewer/components/App/types";
 import { ViewMode } from "../../src/aics-image-viewer/shared/enums";
-import { ViewerChannelSettings } from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
+import {
+  ViewerChannelSetting,
+  ViewerChannelSettings,
+} from "../../src/aics-image-viewer/shared/utils/viewerChannelSettings";
 
-const paramKeys = ["mask", "ch", "luts", "colors", "url", "file", "dataset", "id", "view"];
-type ParamKeys = (typeof paramKeys)[number];
-type Params = { [_ in ParamKeys]?: string };
+const CHANNEL_STATE_KEY_REGEX = /^c[0-9]+$/;
+// Match colon-separated pairs of alphanumeric strings
+const LUT_REGEX = /^[a-z0-9.]*:[ ]*[a-z0-9.]*$/;
+const HEX_COLOR_REGEX = /^[0-9a-fA-F]{6}$/;
 
-function urlSearchParamsToParams(searchParams: URLSearchParams): Params {
+export type ViewerChannelSettingJson = {
+  /** Color, as a 6-digit hex color */
+  col?: string;
+  /** Colorize. "1" is enabled. */
+  clz?: "1" | "0";
+  /** Colorize alpha, in the [0, 1] range. */
+  cza?: string;
+  /** Opacity, in the [0, 1 range] */
+  isa?: string;
+  /** LUT to map from intensity to opacity. Should be two alphanumeric values separated
+   * by a colon, e.g. "0:255", "p50:p90", or "autoij:0".
+   */
+  lut?: string;
+  /** Volume enabled. "1" is enabled. */
+  ven?: "1" | "0";
+  /** Isosurface enabled. "1" is enabled. */
+  sen?: "1" | "0";
+  /** Isosurface value, in the [0, 255] range. */
+  isv?: string;
+};
+
+const baseParamKeys = ["mask", "url", "file", "dataset", "id", "view"] as const;
+const deprecatedParamKeys = ["ch", "luts", "colors"] as const;
+
+type BaseParamKeys = (typeof baseParamKeys)[number];
+type ChannelKey = `c${number}`;
+type DeprecatedParamKeys = (typeof deprecatedParamKeys)[number];
+
+type ParamKeysV0 = BaseParamKeys | DeprecatedParamKeys;
+type ParamKeysV2_9_0 = BaseParamKeys | ChannelKey;
+
+type AllParamKeys = ParamKeysV0 | ParamKeysV2_9_0;
+
+type Params = { [_ in AllParamKeys]?: string };
+
+const isParamKey = (key: string): key is BaseParamKeys => baseParamKeys.includes(key as BaseParamKeys);
+const isDeprecatedParamKey = (key: string): key is DeprecatedParamKeys =>
+  deprecatedParamKeys.includes(key as DeprecatedParamKeys);
+const isChannelKey = (key: string): key is ChannelKey => CHANNEL_STATE_KEY_REGEX.test(key);
+
```

Comment:
@frasercl made some changes here to try and explicitly type parameters, but I want to get your input on if there's better ways to handle this.

Reply:
This looks good! The only thing I'd nitpick is whether `ParamKeysV0` and `ParamKeysV2_9_0` are necessary, or whether they could be documentation. Looks like they're both only used in the definition of `AllParamKeys`, which could just be `BaseParamKeys | DeprecatedParamKeys | ChannelKey`.